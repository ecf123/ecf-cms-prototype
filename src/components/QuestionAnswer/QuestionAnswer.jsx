import "./QuestionAnswer.scss";
import Short from "../Short/Short";
import CheckboxInput from "../CheckboxInput/CheckboxInput";

const QuestionAnswer = ({
  handleShortValue,
  onSubmit,
  handleCheckboxValue,
}) => {
  return (
    <form className="form" onSubmit={onSubmit}>
      <Short
        shortLabelText={"Question"}
        inputClassName="short__input"
        handleShortValue={handleShortValue}
      />
      <p className="form__answers__heading">Answers</p>
      <div className="form__answers">
        <CheckboxInput handleCheckboxValue={handleCheckboxValue} />
        <CheckboxInput handleCheckboxValue={handleCheckboxValue} />
        <CheckboxInput handleCheckboxValue={handleCheckboxValue} />
        <CheckboxInput handleCheckboxValue={handleCheckboxValue} />
      </div>
    </form>
  );
};

export default QuestionAnswer;
