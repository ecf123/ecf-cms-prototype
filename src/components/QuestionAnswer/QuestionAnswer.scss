@use "../../styles/_reset.scss" as *;
@use "../../styles/_variables.scss" as *;

.form {
  height: 22.625rem;
  width: 32rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  &__answers__heading {
    color: $color-blue-navy-dark;
  }
}
