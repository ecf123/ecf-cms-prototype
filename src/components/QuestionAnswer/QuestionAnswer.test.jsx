import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { customRender } from "../../utils/testUtils";

import QuestionAnswer from "./QuestionAnswer";

it("should render the question and answer form", () => {
    const {container} = customRender (<QuestionAnswer />);
    expect(container).toMatchSnapshot();
})

it("should submit the form", () => {
    const onSubmit = jest.fn();
    const { getByText} = render(<QuestionAnswer onSubmit={onSubmit} />);
    fireEvent.submit(getByText("Answers"));
    expect(onSubmit).toHaveBeenCalled();
});