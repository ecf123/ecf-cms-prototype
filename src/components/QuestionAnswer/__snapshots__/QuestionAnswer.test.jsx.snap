// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the question and answer form 1`] = `
<div>
  <form
    class="form"
  >
    <div
      class="short"
    >
      <label
        class="short__label"
      >
        Question
      </label>
      <input
        class="short__input"
        spellcheck="true"
      />
    </div>
    <p
      class="form__answers__heading"
    >
      Answers
    </p>
    <div
      class="form__answers"
    >
      <div
        class="checkbox"
      >
        <input
          class="checkbox__input"
          type="text"
        />
        <input
          type="checkbox"
        />
      </div>
      <div
        class="checkbox"
      >
        <input
          class="checkbox__input"
          type="text"
        />
        <input
          type="checkbox"
        />
      </div>
      <div
        class="checkbox"
      >
        <input
          class="checkbox__input"
          type="text"
        />
        <input
          type="checkbox"
        />
      </div>
      <div
        class="checkbox"
      >
        <input
          class="checkbox__input"
          type="text"
        />
        <input
          type="checkbox"
        />
      </div>
    </div>
  </form>
</div>
`;
